boa_engine/src/builtins/eval/mod.rs

```diff
@@ -257,7 +268,213 @@ impl Eval {
                 context.vm.environments.current_compile_environment(),
                 context,
             );
-            compiler.compile_statement_list_with_new_declarative(&body, true, strict);
+
+            compiler.push_compile_environment(strict);
+            let push_env =
+                compiler.emit_opcode_with_two_operands(Opcode::PushDeclarativeEnvironment);
+
+            let var_environment_is_global = compiler
```

Comment:
This seems like it's the https://tc39.es/ecma262/#sec-evaldeclarationinstantiation operation. Can you document it with the corresponding steps, just for future reference?